### PR TITLE
DA_Week4_복습과제_박보영

### PR DESCRIPTION
**Pull request by bboyeong**
: )
스태킹 앙상블 파트 제대로 풀지 못했습니다ㅠ 피드백에서 알려주시면 감사하겠습니다!!

### CODE REVIEW

**Comment by JeeeWoon** [comment]
안녕하세요 보영님! 우선 과제양이 많았을텐데 열심히 풀어주셔서 감사합니다 :D

1. 우선 5번 문제에서 Encoding 같은 경우엔 잘 아시겠지만 X_wine['type'] = pd.get_dummies(X_wine['type'])로 간단하게 encoding이 가능합니다!
2. xgboost의 f1스코어를 출력할 때 본 데이터셋은 다중분류라 average 옵션을 바꿔주셨는데 관련 링크 첨부합니다! :) (참고 링크: https://aimb.tistory.com/152)
3. 스태킹 관련 코드 첨부합니다! 참고하세요 :)
import numpy as np

from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import AdaBoostClassifier
from sklearn.tree import DecisionTreeClassifier
from lightgbm import LGBMClassifier
from sklearn.linear_model import LogisticRegression

# 개별 ML 모델을 위한 Classifier 생성.
knn_clf  = KNeighborsClassifier(n_neighbors=4)
rf_clf = RandomForestClassifier(n_estimators=100,random_state=0)
dt_clf = DecisionTreeClassifier()
ada_clf = AdaBoostClassifier(n_estimators=100)
lgbm_clf = LGBMClassifier(n_estimators=400)

# 최종 Stacking 모델을 위한 Classifier생성. 
lr_final = LogisticRegression(C=10)

knn_train, knn_test = get_stacking_base_datasets(knn_clf, X_train, y_train, X_test, 7)
rf_train, rf_test = get_stacking_base_datasets(rf_clf, X_train, y_train, X_test, 7)
dt_train, dt_test = get_stacking_base_datasets(dt_clf, X_train, y_train, X_test,  7)    
ada_train, ada_test = get_stacking_base_datasets(ada_clf, X_train, y_train, X_test, 7)
lgbm_train, lgbm_test = get_stacking_base_datasets(lgbm_clf, X_train, y_train, X_test, 7)

Stack_final_X_train = np.concatenate((knn_train, rf_train, dt_train, ada_train, lgbm_train), axis=1)
Stack_final_X_test = np.concatenate((knn_test, rf_test, dt_test, ada_test, lgbm_test), axis=1)
# 잘 진행되었는지 shape 확인
print('원본 학습 피처 데이터 Shape:',X_train.shape, '원본 테스트 피처 Shape:',X_test.shape)
print('스태킹 학습 피처 데이터 Shape:', Stack_final_X_train.shape,
      '스태킹 테스트 피처 데이터 Shape:',Stack_final_X_test.shape)

lr_final.fit(Stack_final_X_train, y_train)
stack_final = lr_final.predict(Stack_final_X_test)

print('최종 메타 모델의 예측 정확도: {0:.4f}'.format(accuracy_score(y_test, stack_final)))
